modules/Bio/EnsEMBL/Compara/RunnableDB/GenomeDBFactory.pm

```diff
@@ -121,6 +123,21 @@ sub fetch_input {
     $genome_dbs = [grep {not $_->genome_component} @$genome_dbs] if not $self->param('component_genomes');
     $genome_dbs = [grep {($_->name eq 'ancestral_sequences') or $_->is_polyploid or $_->genome_component} @$genome_dbs] if not $self->param('normal_genomes');
 
+
+    # filter out exclude_species
+    my %exclude_species;
+    my $_exclude_species = $self->param('exclude_species');
+    if ( ref $_exclude_species eq 'ARRAY' ) {
+        @exclude_species{ @$_exclude_species } = ();
+    } else {
+        @exclude_species{ split(/\s+/, $_exclude_species) } = ();
```

Comment:
Perhaps this has been discussed as a necessary use-case - but I otherwise don't think this needs to be compatible with a space separated string at all. Was this specified somewhere?

Reply:
If space separated strings were not specified to be expected, a more common use-case would be to pass in a comma-separated list, or simply restrict it to either arrays or a single string variable for a single genome to exclude.

Reply:
It was something discussed with Jorge and he linked the example of some code in `Utils`. I just lifted it pretty directly so it may be that a csv string is better?

Reply:
It might make sense in some `Utils` code, but I think parameter wise for a hive runnable, passing just an array would be enough, and this code could be simplified - it would also reduce confusion with the variable names as a bonus.
I would approve based on a change to csv  rather than space separated - because I could see this being used more on the command line when initialising pipeline

Reply:
Have changed to split on a comma in subsequent commits